designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ManagementResources.scala

```diff
@@ -162,7 +162,7 @@ class ManagementResources(val processAuthorizer: AuthorizeProcess,
                 measureTime("cancel", metricRegistry) {
                   processService
                     .cancelProcess(processId, deploymentComment)
```

Comment:
some time ago we introduced a change that deployProcessAsync has two steps:
- validations
- real deploy
It was done this way, because we want to keep modal open until validation passes - it handles 90% of errors. The second part is longer and we don't want that user will wait for that. So we have somewhere in signatures `Future[Future[_]]`. 
I want to "undrescore" in name that the method doesn't wait for deploy finished, but only for validations step. The second part is done async. So it is not about the fact, that we have a Future in signature, but about the fact that we have Future of Future :)
Cancels are done otherwise - user wait for whole process finish.

Reply:
it very valuable explanation, can you also put it in the javadoc of this method?

Reply:
There is Maciek's comment:
> inner Future in result allows to wait for deployment finish, while outer handles validation
, but I can extend it :)

Reply:
done